### ISSUE TITLE
Can not embed into C++ program (dyld: lazy symbol binding failed: Symbol not found: __ZN4node22InitializeNode...)

### ISSUE THREAD

**Issue by buahaha** [nodejs/help#3509]
* **Node.js Version**: v16.6.2
* **OS**: MacOS 11.5.2
* **Scope (install, code, runtime, meta, other?)**: embedding into c++
* **Module (and version) (if relevant)**: node.h

Hey Node devs and users, I have a little tricky situation. I want to embed scripting capabilities into an app (c++), but I have started with simple test: 

```shell
FAIL: //test:node_test (see /private/var/tmp/_bazel_neosb/e834e65c3e7053cdbf2a63d1622160b3/execroot/__main__/bazel-out/darwin-fastbuild/testlogs/test/node_test/test.log)
INFO: From Testing //test:node_test:
==================== Test output for //test:node_test:
dyld: lazy symbol binding failed: Symbol not found: __ZN4node22InitializeNodeWithArgsEPNSt3__16vectorINS0_12basic_stringIcNS0_11char_traitsIcEENS0_9allocatorIcEEEENS5_IS7_EEEESA_SA_
  Referenced from: /private/var/tmp/_bazel_neosb/e834e65c3e7053cdbf2a63d1622160b3/sandbox/darwin-sandbox/89/execroot/__main__/bazel-out/darwin-fastbuild/bin/test/node_test.runfiles/__main__/test/node_test
  Expected in: flat namespace

dyld: Symbol not found: __ZN4node22InitializeNodeWithArgsEPNSt3__16vectorINS0_12basic_stringIcNS0_11char_traitsIcEENS0_9allocatorIcEEEENS5_IS7_EEEESA_SA_
  Referenced from: /private/var/tmp/_bazel_neosb/e834e65c3e7053cdbf2a63d1622160b3/sandbox/darwin-sandbox/89/execroot/__main__/bazel-out/darwin-fastbuild/bin/test/node_test.runfiles/__main__/test/node_test
  Expected in: flat namespace

================================================================================
```

What is the `dyld: Symbol not found: __ZN4node22InitializeNodeWithArgsEPNSt3__16vectorINS0_12basic_stringIcNS0_11char_traitsIcEENS0_9allocatorIcEEEENS5_IS7_EEEESA_SA_` expected library to include and also this symbol's `__ZN4node22InitializeNodeWithArgsEPNSt3__16vectorINS0_12basic_stringIcNS0_11char_traitsIcEENS0_9allocatorIcEEEENS5_IS7_EEEESA_SA_` library needed in c++ binary...

In other words, what is the missing component to run this using `bazel build //test:node_test`?

[My repo](https://github.com/buahaha/game-cpp_pro)

**Comment by addaleax** [comment]
Well, this seems more like a bazel question and not a Node.js one, but generally, the problem here is that you don’t seem to be linking in Node.js as a library.

**Comment by buahaha** [comment]
let me rephrase my question. what `should` I include?
